testing/integration/diagnostics_test.go

```diff
@@ -139,28 +139,44 @@ func TestDiagnosticsCommand(t *testing.T) {
 
 func testDiagnosticsFactory(t *testing.T, diagFiles []string, diagCompFiles []string, fix *integrationtest.Fixture, cmd []string) func(ctx context.Context) error {
 	return func(ctx context.Context) error {
-		diagnosticCommandWD := t.TempDir()
-		diagnosticCmdOutput, err := fix.Exec(ctx, cmd, process.WithWorkDir(diagnosticCommandWD))
-
-		t.Logf("diagnostic command completed with output \n%q\n", diagnosticCmdOutput)
-		require.NoErrorf(t, err, "error running diagnostic command: %v", err)
-
-		t.Logf("checking directory %q for the generated archive", diagnosticCommandWD)
-		files, err := filepath.Glob(filepath.Join(diagnosticCommandWD, diagnosticsArchiveGlobPattern))
-		require.NoError(t, err)
-		require.Len(t, files, 1)
-		t.Logf("Found %q diagnostic archive.", files[0])
+		diagZip, err := collectDiagnostics(ctx, t, fix, cmd...)
 
 		// get the version of the running agent
 		avi, err := getRunningAgentVersion(ctx, fix)
 		require.NoError(t, err)
 
-		verifyDiagnosticArchive(t, files[0], diagFiles, diagCompFiles, avi)
+		verifyDiagnosticArchive(t, diagZip, diagFiles, diagCompFiles, avi)
 
 		return nil
 	}
 }
 
+// collectDiagnostics collects the agent diagnostics and returns the path to the
```

Comment:
Nit: this something shared among different integration tests in different files, maybe we want to move it to a common utilities file